src/uu/dd/src/dd.rs

```diff
@@ -0,0 +1,1296 @@
+            records_truncated: rstat.records_truncated,
+            duration: start.elapsed(),
+        })?;
+    }
+
+    if o.cflags.fsync {
+        o.fsync()?;
+    } else if o.cflags.fdatasync {
+        o.fdatasync()?;
+    }
+
+    match i.xfer_stats {
+        Some(StatusLevel::Noxfer) | Some(StatusLevel::None) => {}
+        _ => print_xfer_stats(&ProgUpdate {
+            reads_complete: rstat.reads_complete,
+            reads_partial: rstat.reads_partial,
+            writes_complete: wstat.writes_complete,
+            writes_partial: wstat.writes_partial,
+            bytes_total: wstat.bytes_total,
+            records_truncated: rstat.records_truncated,
+            duration: start.elapsed(),
+        }),
+    }
+    Ok(())
+}
+
+// The compiler does not like Clippy's suggestion to use &str in place of &String here.
+#[allow(clippy::ptr_arg)]
+fn append_dashes_if_not_present(mut acc: Vec<String>, s: &String) -> Vec<String> {
+    if Some("--") != s.get(0..=1) {
+        acc.push(format!("--{}", s));
+    }
+    acc
+}
+
+macro_rules! unpack_or_rtn (
+    ($i:expr, $o:expr) =>
+    {{
+        match ($i, $o)
+        {
+            (Ok(i), Ok(o)) =>
+                (i,o),
+            (Err(e), _) =>
+            {
+                eprintln!("dd Error: {}", e);
+                return RTN_FAILURE;
+            },
+            (_, Err(e)) =>
+            {
+                eprintln!("dd Error: {}", e);
+                return RTN_FAILURE;
+            },
+        }
+    }};
+);
+
+pub fn uumain(args: impl uucore::Args) -> i32 {
+    let dashed_args = args
+        .collect_str(InvalidEncodingHandling::Ignore)
+        .accept_any()
+        .iter()
+        .fold(Vec::new(), append_dashes_if_not_present);
+
+    let matches = uu_app()
+        // TODO: usage, after_help
+        //.usage(...)
+        //.after_help(...)
+        .get_matches_from(dashed_args);
+
+    let result = match (
+        matches.is_present(options::INFILE),
+        matches.is_present(options::OUTFILE),
+    ) {
+        (true, true) => {
+            let (i, o) =
+                unpack_or_rtn!(Input::<File>::new(&matches), Output::<File>::new(&matches));
+
+            dd_fileout(i, o)
+        }
+        (false, true) => {
+            let (i, o) = unpack_or_rtn!(
+                Input::<io::Stdin>::new(&matches),
+                Output::<File>::new(&matches)
+            );
+
+            dd_fileout(i, o)
+        }
+        (true, false) => {
+            let (i, o) = unpack_or_rtn!(
+                Input::<File>::new(&matches),
+                Output::<io::Stdout>::new(&matches)
+            );
+
+            dd_stdout(i, o)
+        }
+        (false, false) => {
+            let (i, o) = unpack_or_rtn!(
+                Input::<io::Stdin>::new(&matches),
+                Output::<io::Stdout>::new(&matches)
+            );
+
+            dd_stdout(i, o)
+        }
+    };
+    match result {
+        Ok(_) => RTN_SUCCESS,
+        Err(e) => {
```

Comment:
Same as with the other `debug_println!`, this warning is because `e` is unused in release mode.